### DISCUSSION TITLE
Arkana fails on CI due to .env file missing

### DISCUSSION THREAD

**Discussion by mishaherasimov** [rogerluan/arkana#32] [Q&A] [answered]
I'm currently trying to build a pipeline that creates keys before building the project. 
I've imported all keys as environmental variables of the project.

Given these conditions, Arkana fails to generate an SPM package because .env is added to .gitignore.
What would be the correct way to configure Arkana?

CI Environment: Circle CI

<img width="845" alt="Screenshot 2023-03-17 at 2 46 38 PM" src="https://user-images.githubusercontent.com/20767940/225992079-96b3f8c8-deeb-4679-8fd0-574657e65a8c.png">

**Comment by rogerluan** [comment]
Hi @mishaherasimov 👋  sorry for the late response

Do I understand correctly that you don't want to use CI env vars in your pipeline? Thus, sounds like you want to fetch the env vars during the CI pipeline execution from a remote location, and save them somewhere to be picked up by Arkana, is that correct?

>because .env is added to .gitignore.

Would uncommenting your gitignore .env config a valid solution in your case?

Alternatively, you could also export your secrets using `export KEY_NAME = "key_value"`, or by running `source .env` (if you populated your `.env` using the `export …` syntax I just said. Would this work?

Looking forward to hearing more from you 🙏

**Reply by mishaherasimov** [reply]
Thank you for your reply @rogerluan,

Arkana documentation suggests excluding `.env` file from the repository to avoid compromising the keys. I did that and imported all environmental variables to the Circle CI project configuration.

However, when I run `arkana`, it fails on CI because it can't find the `.env` file. I was expecting that the program would use sourced environmental variables instead (if they are present in the environment) without requiring a specific path to `.env` file.

I resolved this issue by eventually committing `.env` file to the repository and providing direct mapping of environmental variables e.g.

```
MyPropertyProd = ${MY_PROPERTY_CIRCLE_CI_PROD}
MyPropertyStag = ${MY_PROPERTY_CIRCLE_CI_STAG}
```

It would be nice to have the option to use the environment as that source of values instead of a specific `.env` file:
e.g. `bundle exec arkana --use-local-env`

**Reply by rogerluan** [reply]
Ooh, got it! Now it's more clear 🙏 thanks for clarifying!

So yeah, it definitely sounds like a bug, I'll get this fixed.

>Arkana documentation suggests excluding .env file from the repository to avoid compromising the keys.

This is 100% accurate, my suggestion was based on the fact that I thought that you were pulling the secrets from a remote server and populating them in a locally-generated .env file. But nevermind what I said 😄 your report now makes total sense.

Thanks fore reporting this 💪

**Comment by mishaherasimov** [comment]
Thank you for adding it to the list of issues!

**Comment by rogerluan** [comment, marked as answer]
This issue has been fixed already, see: https://github.com/rogerluan/arkana/issues/33#issuecomment-2061781110
